packages/turbo-codemod/src/transforms/migrate-dot-env.ts

```diff
@@ -0,0 +1,103 @@
+import path from "node:path";
+import { readJsonSync, existsSync } from "fs-extra";
+import { type PackageJson, getTurboConfigs } from "@turbo/utils";
+import type { Schema as TurboJsonSchema } from "@turbo/types";
+import type { Transformer, TransformerArgs } from "../types";
+import { getTransformerHelpers } from "../utils/getTransformerHelpers";
+import type { TransformerResults } from "../runner";
+
+// transformer details
+const TRANSFORMER = "migrate-dot-env";
+const DESCRIPTION = 'Migrate the "dotEnv" entries to "inputs" in `turbo.json`';
+const INTRODUCED_IN = "2.0.0";
+
+function migrateConfig(config: TurboJsonSchema) {
+  if ("globalDotEnv" in config) {
+    if (config.globalDotEnv) {
+      config.globalDependencies = config.globalDependencies ?? [];
+      for (const dotEnvPath of config.globalDotEnv) {
+        config.globalDependencies.push(dotEnvPath);
+      }
+    }
+    delete config.globalDotEnv;
+  }
+
+  for (const [_, taskDef] of Object.entries(config.pipeline)) {
```

Comment:
Not sure how, but the `pipeline` -> `tasks` codemod should run first and then this is just `config.tasks`. 

Almost like we need a `dependsOn` config for codemods 😓